### PR TITLE
Fix remaining Rust 1.50 clippy issues (SDK-220)

### PR DESCRIPTION
**Pull request by dmose**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `automation/all_tests.sh` runs to completion and produces no failures
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
  - _no new functionality here, just making code pass clippy lints_
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
  - _No new functionality_
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: components/support/viaduct-reqwest/src/lib.rs [hunk 1]
```diff
@@ -35,6 +35,7 @@ lazy_static::lazy_static! {
     };
 }
 
+#[allow(clippy::unnecessary_wraps)] // not worth the time to untangle
```
Comment (rfk) [inline components/support/viaduct-reqwest/src/lib.rs:38]: ➕ 1️⃣

File: components/sync15/src/migrate_state.rs [hunk 2]
```diff
@@ -123,6 +123,7 @@ mod tests {
         get_state_with_engine_changes_and_declined(changes, "")
     }
 
+    #[allow(clippy::unnecessary_wraps)]
```
Comment (rfk) [inline components/sync15/src/migrate_state.rs:126]: nit: I like the way you had comments on other instances of `allow`, please consider adding one here as well.

**Review by rfk — APPROVED** [review]
L.G.T.M. Thanks @dmose!

In theory, you should have sufficient privileges on this repo to merge this, so I'm going to leave the final merge to you in order to test out that theory :-)

File: components/sync15/src/migrate_state.rs (hunk 2 above)
Comment (dmose) [reply components/sync15/src/migrate_state.rs]: done.

**Review by rfk — APPROVED** [review]
rubber-stamp re-r+ in open defiance of mergify bot dismissing my previous r+

**Pull request merged**